pkg/operator/controller/controller.go

```diff
@@ -171,6 +207,39 @@ func (r *ReconcileCDI) Reconcile(request reconcile.Request) (reconcile.Result, e
 	return r.reconcileUpdate(reqLogger, cr)
 }
 
+func shouldTakeUpdatePath(targetVersion, currentVersion string) (bool, error) {
+
+	// if no current version, then this can't be an update
+	if currentVersion == "" {
+		return false, nil
+	}
+
+	if targetVersion == currentVersion {
+		return false, nil
+	}
+
+	// semver doesn't like the 'v' prefix
+	targetVersion = strings.TrimPrefix(targetVersion, "v")
+	currentVersion = strings.TrimPrefix(currentVersion, "v")
+
+	// our default position is that this is an update.
+	// So if the target and current version do not
+	// adhere to the semver spec, we assume by default the
+	// update path is the correct path.
+	shouldTakeUpdatePath := true
+	target, err := semver.Make(targetVersion)
+	if err == nil {
+		current, err := semver.Make(currentVersion)
+		if err == nil {
+			if target.Compare(current) <= 0 {
+				shouldTakeUpdatePath = false
+			}
+		}
+	}
+
```

Comment:
Yes

Reply:
I recommend a more forceful approach. I imagine this is your own versioning scheme and there is no reason for it not to follow the semver pattern. If it does not, you should do error logs and panic. Somebody has messed up the version, and better to find out during unit tests or integration tests, then realize months later that you've potentially implemented a downgrade feature that doesn't work, or that you are otherwise "upgrading" when you shouldn't.

Reply:
Anything that gets released will be semver tagged.  

And if the version is downgraded, the operator will not do any reconciling.  IMO, this is better than panicking because the user will still be allowed to delete the CDI CR (uninstall).  And that is not possible if the operator is not running (CR has a finalizer).

As for supporting only semver tags, maybe in the future.  I think I like the idea of always being to upgrade to "latest".  Or some other explicit tag.